test/sql/cast/test_bit_cast.test

```diff
@@ -0,0 +1,216 @@
+# name: test/sql/cast/test_bit_cast.test
+# description: Test casting from and to BitString
+# group: [cast]
+
```

Comment:
Can we add some tests for all the min/max values for the numeric types, casting to and from bit?